core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala

```diff
@@ -64,8 +64,8 @@ class JdbcRDD[T: ClassTag](
     // bounds are inclusive, hence the + 1 here and - 1 on end
     val length = 1 + upperBound - lowerBound
     (0 until numPartitions).map(i => {
-      val start = lowerBound + ((i * length) / numPartitions).toLong
-      val end = lowerBound + (((i + 1) * length) / numPartitions).toLong - 1
+      val start = lowerBound + ((BigDecimal(i) * length) / numPartitions).toLong
```

Comment:
@srowen Sorry I don't understand how last partition can get >= upperBound

With this codes

val length = 1 + upperBound - lowerBound
(0 until numPartitions).map(i => {
      val start = lowerBound + ((BigInt(i) * length) / numPartitions).toLong
      val end = lowerBound + ((BigInt(i + 1) * length) / numPartitions).toLong - 1
      new JdbcPartition(i, start, end)
 })

the last iteration on this:
i = numPartitions - 1

So end = lowerBound + (numPartitions - 1 + 1) * (1 + upperBound - lowerBound) / numPartitions - 1
= lowerBound + numPartitions/ numPartitions * (1 + upperBound - lowerBound)  - 1
= lowerBound 1 + upperBound - lowerBound - 1
= upperBound